src/plugins/intel_cpu/src/nodes/if.cpp

```diff
@@ -193,7 +207,15 @@ void If::prepareBeforeMappers(const bool isThen, const dnnl::engine& eng) {
     auto &beforeMappers = isThen ? beforeThenMappers : beforeElseMappers;
     for (auto& map_rule : inputPortMap) {
         auto fromMem = getParentEdgesAtPort(map_rule.from)[0]->getMemoryPtr();
-        auto &toMems = inputMems[map_rule.to];
+        auto& toMems = inputMems[map_rule.to];
+        // Check precision between If node input/output and it's subgrapsh input/output.
+        for (const auto& toMem : toMems) {
+            if (fromMem->getDesc().getPrecision() != toMem->getDesc().getPrecision()) {
+                OPENVINO_WARN << "If node fromMem and toMem precision mismatch: from "
+                              << fromMem->getDesc().getPrecision().to_string() << " to "
+                              << toMem->getDesc().getPrecision().to_string();
+            }
+        }
```

Comment:
Do we really need it in a production code? I assume, users are not interested in this information 😄 

I think it's probably may be useful in debugging when `DEBUG_CAPS=ON`. If it's so, I suggest to use our internal macros `DEBUG_LOG(...);` instead of `OPENVINO_WARN`
What do you think?

Reply:
Updated